vars/sonarExecuteScan.groovy

```diff
@@ -191,3 +206,30 @@ private void loadSonarScanner(config){
         mv ${foldername} .sonar-scanner
     """
 }
+
+private void loadCertificates(Map config) {
+    String certificateFolder = '.certificates/'
+    List wgetOptions = [
+        "--directory-prefix ${certificateFolder}"
+    ]
+    List keytoolOptions = [
+        '-import',
+        '-noprompt',
+        '-storepass changeit',
+        '-keystore .sonar-scanner/jre/lib/security/cacerts'
+    ]
+    if (config.customTlsCertificateLinks){
+        if(config.verbose){
+            wgetOptions.push('--verbose')
+            keytoolOptions.push('-v')
+        }else{
+            wgetOptions.push('--no-verbose')
+        }
+        config.customTlsCertificateLinks.each { url ->
+            def filename = new File(url).getName()
+            filename = URLDecoder.decode(filename, java.nio.charset.StandardCharsets.UTF_8.name())
+            sh "wget ${wgetOptions.join(' ')} ${url}"
```

Comment:
[clarification] Normally we use `curl` inside the piper lib. I guess we have `wget` inside the image used here? ... But above we use `curl`. That is strange, maybe we should stick to `curl`.

How does `wget` behave in case of 4xx and 5xx responses? Does this result in a returnCode !=0? In case not: we continue and fail later instead of directly failing here (which makes troubleshooting easier)

Reply:
This was the simplest solution (copy&paste).